src/prefect/orion/schemas/states.py

```diff
@@ -52,11 +54,37 @@ class StateDetails(PrefectBaseModel):
     refresh_cache: bool = None
 
 
-class State(IDBaseModel, Generic[R]):
+class StateBaseModel(IDBaseModel):
+    def orm_dict(
+        self, *args, shallow: bool = False, json_compatible: bool = False, **kwargs
+    ) -> dict:
+        schema_dict = self.dict(
+            *args, shallow=shallow, json_compatible=json_compatible, **kwargs
+        )
+        # remove the data field in order to construct a state ORM model
+        schema_dict.pop("data", None)
+        return schema_dict
+
+
+class StateGetter(GetterDict):
+    def get(self, key: str, default: Any) -> Any:
+        # During orchestration, ORM states can be instantiated prior to inserting
+        # results into the artifact table, and the `data` field will fail to load
+        if key == "data":
+            try:
+                return super().get(key, default)
+            except MissingGreenlet:
+                return None
```

Comment:
This seems dangerous — can we expand on why we would swallow this error?

Reply:
there's a comment above and we plan to add tests, do we need more clarity than that?

```
# During orchestration, ORM states can be instantiated prior to inserting
# results into the artifact table, and the `data` field will fail to load
```

Reply:
I've updated the comment to the following:

```
# During orchestration, ORM states can be instantiated prior to inserting
# results into the artifact table and the `data` field will not be eagerly
# loaded. In these cases, sqlalchemy will attept to lazily load the the
# relationship, which will fail when called within a synchronous pydantic
 # method
```

Reply:
That's better, I think. It does not necessarily cover why it doesn't matter that we can't load the field though.

That this is necessary at all worries me and could hide a bug in the future, is there a way we can avoid making the call that causes the problem in the first place?

Reply:
I think it will be brittle; we'd have to avoid using the `from_orm` method on the pydantic model, which means we'll have to stitch the object together by hand. I think this is prone to falling out of sync with our state ORM/schema implementations